module/mempool/stdmap/backdata/arrayBackData.go

```diff
@@ -0,0 +1,400 @@
+package backdata
+
+import (
+	"encoding/binary"
+	"time"
+	_ "unsafe" // for linking runtimeNano
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/mempool/stdmap/backdata/heropool"
+)
+
+//go:linkname runtimeNano runtime.nanotime
+func runtimeNano() int64
+
+const (
+	bucketSize = uint64(16)
+
+	// telemetryCounterInterval is the number of required interactions with
+	// this back data prior to printing any log. This is done as a slow-down mechanism
+	// to avoid spamming logs upon read/write heavy operations. An interaction can be
+	// a read or write.
+	telemetryCounterInterval = uint64(1000)
+
+	// telemetryDurationInterval is the required elapsed duration interval
+	// prior to printing any log. This is done as a slow-down mechanism
+	// to avoid spamming logs upon read/write heavy operations.
+	telemetryDurationInterval = 10 * time.Second
+)
+
+// bIndex is data type representing a bucket index.
+type bIndex uint64
+
+// sIndex is data type representing a slot index in a bucket.
+type sIndex uint64
+
+// idPrefix is a 32-bits prefix flow.Identifier used to determine the bIndex of the entity
+// it represents.
+type idPrefix uint32
+
+type key struct {
+	keyIndex    uint64          // slot age.
+	entityIndex heropool.EIndex // link to actual entity.
```

Comment:
No intention to debate around the suggestion, but `ArrayBackData` implements [`BackData`](https://github.com/onflow/flow-go/blob/6d0227e46d338355609c9cf8429f052479420aa5/module/mempool/backData.go#L7-L40) generic interface of Flow. This interface revolves around the idea of storing and retrieving `entity` objects. Accordingly, `entityIndex` is more intention revealing and adds more integrity than `poolIndex`.

Reply:
@yhassanzadeh13 hmmm... maybe. I'm looking forward to seeing how this works out if/when `ArrayBackData` implements not only the `BackData` generic interface of Flow, but also other interfaces. But I guess we'll cross that bridge when it comes :-)

In the meantime, what about considering simplification in the other direction? Renaming `heroPool` to e.g. `heroEntities` or `heroEntityPool`? Could this be more intention revealing?

I like the word 'pool' because it isn't obvious how you access it like 'slice' or 'array' or 'map'. However, it does imply being quite generic like 'slice' or 'array' or 'map' which might be misleading if heroPool is not generic and instead specifically orientated towards 'entity objects'? In which case maybe `heroEntityPool` is the right amount of generic and intention revealing?

As always... naming is the most difficult thing... :-)